packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -951,6 +1029,12 @@ export class AttributesController extends ConsumptionBaseController {
         }
 
         const ownSharedIdentityAttributes = await this.getLocalAttributes(query);
+        try {
+            ownSharedIdentityAttributes.sort((a, b) => (a.shareInfo!.peer < b.shareInfo!.peer ? 1 : -1));
+        } catch (e: unknown) {
+            throw CoreErrors.attributes.cannotSortAttributesByPeers(e);
+        }
```

Comment:
sort can throw an error?

Reply:
ah, make sure to use ? instead of !

Reply:
This gives an error, since `a.shareInfo.peer` might be undefined.

Reply:
yeah, you have to ignore if one is undefined. `!` throws errors so you only may use it when you are **totally** sure that it will not be undefined (e.g. when you have checked it before but the compiler doesn't knows that). In this case you are not sure so it throws an error. This should not happen here tbh.

Reply:
Also I meant to share the attributes by creation date instead of sorting it by the peer. IMO it doesn't make much sense to do this.

Reply:
Ok, I thought sorting them by peer would give some kind of consistency throughout the versions. But I don't mind sorting them be their creation date.

Reply:
maybe one other thing: this is one of the best examples where we should **not** use error codes. This is nothing the user has caused, just some internal stuff that's going wrong.

Reply:
You can use `expect(foo).toStrictEqual(expect.arrayContaining([bar, baz]));` to ignore the order. Please try this before.

Reply:
This seems to be working. So, are you fine the returned result is not sorted?

Reply:
yeah, sure.